ccmlib/common.py

```diff
@@ -854,20 +854,28 @@ def _get_jdk_version(version):
     ver_pattern = '\"(\d+).*\"'
     return re.search(ver_pattern, str(version)).groups()[0] + ".0"
 
+def get_supported_jdk_versions_internal(path, pattern):
+    if os.path.exists(path):
+        with open(path) as f:
+            for line in f:
+                match = re.search(pattern, line)
+                if match:
+                    versions = match.group(1).split(',')
+                    versions = [8 if v == '1.8' else int(v) for v in versions]
+                    return versions
+    return None
+
 def get_supported_jdk_versions(install_dir):
     """
-    Return the supported java versions from build.xml
+    Return the supported java versions from build.xml or from cassandra.in.sh
     Only works in > 4.1
     """
-    build = os.path.join(install_dir, 'build.xml')
-    with open(build) as f:
-        for line in f:
-            match = re.search('name="java\.supported" value="([0-9.,]+)"', line)
-            if match:
-                versions = match.group(1).split(',')
-                versions = [8 if v == '1.8' else int(v) for v in versions]
-                return versions
-    raise ValueError("did not find java.supported in build.xml!")
+
+    versions = get_supported_jdk_versions_internal(os.path.join(install_dir, 'build.xml'), 'name="java\.supported" value="([0-9.,]+)"')
+    if versions is None:
+        versions = get_supported_jdk_versions_internal(os.path.join(install_dir, 'bin', 'cassandra.in.sh'), 'java_versions_supported=([0-9.,]+)')
```

Comment:
nit:  the former works for source ccm versions, the latter for binary versions, checking both seems odd.   but maybe this is as effective as it gets … 🤷

Reply:
i was hoping there was some flag that says whether this is a downloaded or built install.

the conditional fall-throw code is not intuitive.

my suggestion is just add a comment…  
```suggestion
    // source built installs from 5.0 have supported java versions specified in build.xml
    versions = get_supported_jdk_versions_internal(os.path.join(install_dir, 'build.xml'), 'name="java\.supported" value="([0-9.,]+)"')
    if versions is None:
        // downloaded binary installs from 5.1 have supported java versions specified in bin/cassandra.in.sh
        versions = get_supported_jdk_versions_internal(os.path.join(install_dir, 'bin', 'cassandra.in.sh'), 'java_versions_supported=([0-9.,]+)')
```